pkg/router/template/util/util.go

```diff
@@ -0,0 +1,71 @@
+package util
```

Comment:
Don't put methods into packages called util.  Is there any reason this isn't in the haproxy package?  Because this looks haproxy specific.

Reply:
Moved those out of the `template_helper.go` code as they are common - used by both nginx and older haproxy templates (the new template would just call those functions directly instead of via the the template_helper code). All expect for the last one `GenerateBackendNamePrefix` which I guess is sorta haproxy specific - but technically just a prefix name, so qualifies as an utility function.